src/components/LoginForm.jsx

```diff
@@ -1,29 +1,38 @@
-import React, { useState } from "react";
-import { post } from "../clients/HttpClient";
-import { Form, FormGroup, Label, Input, Button, Alert } from "reactstrap";
+import React, {useState} from "react";
+import {post} from "../clients/HttpClient";
+import {Alert, Button, Form, FormGroup, Input, Label} from "reactstrap";
+import {useNavigate} from "react-router-dom";
+import {APP_PROPS} from "../constants/AppConstants";
 
 const LoginForm = () => {
   const [email, setEmail] = useState("");
   const [password, setPassword] = useState("");
   const [error, setError] = useState("");
-
+  const pageNavigation = useNavigate();
   const handleSubmit = async (e) => {
     e.preventDefault();
     const onSuccess = (response) => {
       if (response.data.value) {
         localStorage.setItem("token", response.data.value);
+        pageNavigation("/catalog/all");
+        console.log("Login successful:", response.data);
+      } else if (response.data.message === "login failed") {
+        console.log("Login unsuccessful :", response.data);
+        setError("Invalid email or password!");
+      } else {
+        console.log("Unexpected Error", response.data);
+        setError("Unexpected Error occurred")
       }
-      console.log("Login successful:", response.data);
     };
 
     const onError = (error) => {
       console.error("Login error:", error);
       setError("Invalid email or password!");
     };
 
-    const apiUrl = "http://localhost:8081/user/login";
+    const apiUrl = `${APP_PROPS.bookstoreUrl}/user/login`;
     const userData = { email, password };
-    post(apiUrl, userData, onSuccess, onError);
+    await post(apiUrl, userData, onSuccess, onError);
```

Comment:
Why do we need await.

Reply:
since post is an asynchronous function and we wait till it gets executed so that the onsuccess and onError functions get the response properly and they doesn't project any abnormal behaviour